source/server/admin/dynamic_stats.js

```diff
@@ -0,0 +1,88 @@
+// Before first checkin:
+//   * top-n algo to limit compute overhead with high cardinality stats with user control of N.
+//   * alternate sorting criteria, reverse-sort controls, etc.
+
+// Follow-ups:
+//   * render histograms
+//   * detect when user is not looking at page and stop or slow down pinging the server
+//   * hieararchical display
+//   * json flavor to send hierarchical names to save serialization/deserialization costs
+
+let current_stats = new Map;
+
+const param_id_prefix = 'param-1-stats-';
+
+function loadStats() {
+  val = name => name + '=' + document.getElementById(param_id_prefix + name).value;
+  const params = ['usedonly', 'filter', 'type', 'histogram_buckets'];
+  const url = '/stats?format=json&' + params.map(val).join('&');
+  fetch(url, {
+    method: 'GET',
+    headers: {},
+  })
+      .then(response => response.json())
+      .then(data => renderStats(data));
+}
+
+function renderStats(data) {
+  const content = document.getElementById('dynamic-stats');
+  while (content.firstChild) {
```

Comment:
there should be a way to clear a node without iteration

Reply:
Maybe -- but that's what I found when I looked up "how to clear all children from a dom node" :) Maybe a recent-ish addition to the dom model? Or part of jsquery?

I wanted to keep this using vanilla js.

But actually this isn't necessary anyway right now; I was thinking I might have made the output display have some HTML structure but I just wound up throwing it all in a div for now.